include/umf/providers/provider_level_zero.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_PROVIDER_LEVEL_ZERO_H
+#define UMF_PROVIDER_LEVEL_ZERO_H
+
+#include <level_zero/ze_api.h>
```

Comment:
I don't think we can include this header here. Our customers might not have the headers installed, if they use dlopen, right? @vinser52?

Reply:
but how they could pass ze_context_handle_t or ze_device_handle_t to the level zero provider params?

Reply:
They might have a copy of the headers somewhere, but not necessarily in <level_zero/ze_api.h> I think.

Reply:
or I can check the #define and include the file only when the user hadn't done it yet + print the messafe

Reply:
We can also just accept 'void*' for each param - I don't think there is a good solution for that if we want to use dlopen.

Reply:
Ok I removed this header from the interface